### PR TITLE
Reenable MRTCore and Test Stage in Main

### PR DESCRIPTION
**Pull request by kythant**
1. Set Nuget.Config to be public
2. Revert Branch checkout in eng/common/AzurePipelinesTemplates/WindowsAppSDK-BuildSetup-Steps.yml
3. Reenable MRTCore and Test Stage

### CODE REVIEW

File: build/ProjectReunion-CI.yml [hunk 1]
```diff
@@ -117,111 +122,115 @@ jobs:
...
-    inputs:
-      PathtoPublish: '$(Build.SourcesDirectory)\BuildOutput\FullNuget'
-      artifactName: 'NugetContent'
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish artifact: TransportPackage'
-    inputs:
-      PathtoPublish: '$(build.artifactStagingDirectory)\FullNuget'
-      artifactName: 'TransportPackage'
+# Public validation for BuildMRT will be disabled until Microsoft.Taef becomes public.
```
Comment (alexlamtest) [inline build/ProjectReunion-CI.yml:125]: Can you elaborate what you mean by "Microsoft.Taef becomes public"?
My understanding is that TAEF is already public:
https://learn.microsoft.com/en-us/windows-hardware/drivers/taef/

Comment (kythant) [reply build/ProjectReunion-CI.yml:125]: The TAEF package is not available on Nuget.org yet

File: dev/MRTCore/mrt/Core/unittests/packages.config [hunk 2]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Taef" version="10.75.221207001" targetFramework="native" />
+  <package id="Microsoft.Taef" version="10.58.210222006-develop" targetFramework="native" />
```
Comment (alexlamtest) [inline dev/MRTCore/mrt/Core/unittests/packages.config:3]: I see in the PR description that reverting the version of TAEF is intended. But this is a fairly big jump backwards, 20+ months, is this move explained somewhere?
Also I wasn't able to find "10.58.210222006-develop" here: https://www.osgwiki.com/wiki/TAEF_Release_Notes#Fix_Use-after-free_when_Appending_a_String_to_Itself, so I'm unclear why it is picked as opposed to, say, v10.58.210305002 mentioned there.

**Review by alexlamtest — APPROVED** [review]

File: dev/MRTCore/mrt/Core/unittests/packages.config (hunk 2 above)
Comment (kythant) [reply dev/MRTCore/mrt/Core/unittests/packages.config]: 10.58.210222006-develop is no more :)

**Pull request merged**